docs/library/index.rst

```diff
@@ -1,21 +1,239 @@
+    { "name": "pwm_1 offset", "wave": "9............................","data":["1"]},
+    { "name": "pwm_2 period", "wave": "6............................","data":["8"]},
+    { "name": "pwm_2 pulse", "wave": "6............................","data":["3"]},
+    { "name": "pwm_2 offset", "wave": "6............................","data":["5"]},
+    { "name": "external_sync", "wave": "1....0.......................",phase: 0.5,},
+    { "name": "offset counter", "wave": "=.....7777777777777777777777=","data":["0","1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22"]},
+    { "name": "counter 0", "wave": "=......444444455555555444444=","data":["1","2","3","4","5","6","7","8","1","2","3","4","5","6","7","8","1","2","3","4","5","6"]},
+    { "name": "pwm 0", "wave": "l......h..l....h..l....h..l.."},
+    { "name": "counter 1", "wave": "=..........44444445555555544=","data":["1","2","3","4","5","6","7","8","1","2","3","4","5","6","7","8","1","2"]},
+    { "name": "pwm 1", "wave": "l..........h..l....h..l....h."},
+  ],
+  foot: {text: ['tspan', 'External sync, start at sync (default) e.g.'],
+  }}
+  
+.. wavedrom::
 
-Contents
--------------------------------------------------------------------------------
+  { "signal" : [
+    { "name": "clk", "wave": "P............................"},
+    { "name": "pwm_1 period", "wave": "9............................","data":["8"]},
+    { "name": "pwm_1 pulse", "wave": "9............................","data":["3"]},
+    { "name": "pwm_1 offset", "wave": "9............................","data":["1"]},
+    { "name": "pwm_2 period", "wave": "6............................","data":["8"]},
+    { "name": "pwm_2 pulse", "wave": "6............................","data":["3"]},
+    { "name": "pwm_2 offset", "wave": "6............................","data":["5"]},
+    { "name": "external_sync", "wave": "1....0.......................",phase: 0.5,},
+    { "name": "offset counter", "wave": "=.....7777777777777777777777=","data":["0","1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22"]},
+    { "name": "counter 0", "wave": "=......444444455555555444444=","data":["1","2","3","4","5","6","7","8","1","2","3","4","5","6","7","8","1","2","3","4","5","6"]},
+    { "name": "pwm 0", "wave": "l.............h..l....h..l..."},
+    { "name": "counter 1", "wave": "=..........44444445555555544=","data":["1","2","3","4","5","6","7","8","1","2","3","4","5","6","7","8","1","2"]},
+    { "name": "pwm 1", "wave": "l.................h..l....h.."},
+  ],
+  foot: {text: ['tspan', 'External sync without start at sync e.g.'],
+  }}
 
-.. toctree::
-   :maxdepth: 1
+Register Map
+--------------------------------------------------------------------------------
 
-   axi_dmac/index
-   spi_engine/index
-   jesd204/index
-   axi_adxcvr/index
-   xilinx/index
+.. hdl-regmap::
+   :name: axi_pwm_gen
```

Comment:
The Software Guidelines section should be added as well, to point towards the documentation and code for the software support